wire-gson-support/src/main/java/com/squareup/wire/WireTypeAdapterFactory.java

```diff
@@ -13,43 +13,40 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.squareup.wire;
+package com.squareup.wire
 
-import com.google.gson.Gson;
-import com.google.gson.TypeAdapter;
-import com.google.gson.TypeAdapterFactory;
-import com.google.gson.reflect.TypeToken;
-import javax.annotation.Nullable;
-import okio.ByteString;
+import com.google.gson.Gson
+import com.google.gson.TypeAdapter
+import com.google.gson.TypeAdapterFactory
+import com.google.gson.reflect.TypeToken
+import okio.ByteString
 
 /**
- * A {@link TypeAdapterFactory} that allows Wire messages to be serialized and deserialized
- * using the GSON Json library. To create a {@link Gson} instance that works with Wire,
- * use the {@link com.google.gson.GsonBuilder} interface:
+ * A [TypeAdapterFactory] that allows Wire messages to be serialized and deserialized
+ * using the GSON Json library. To create a [Gson] instance that works with Wire,
+ * use the [com.google.gson.GsonBuilder] interface:
  *
- * <pre>
+ * ```
  * Gson gson = new GsonBuilder()
  *     .registerTypeAdapterFactory(new WireTypeAdapterFactory())
  *     .create();
- * </pre>
+ * ```
  *
- * The resulting {@link Gson} instance will be able to serialize and deserialize any Wire
- * {@link Message} type, including extensions and unknown field values. The JSON encoding is
+ * The resulting [Gson] instance will be able to serialize and deserialize any Wire
+ * [Message] type, including extensions and unknown field values. The JSON encoding is
  * intended to be compatible with the
- * <a href="https://code.google.com/p/protobuf-java-format/">protobuf-java-format</a>
+ * [protobuf-java-format](https://code.google.com/p/protobuf-java-format/)
  * library. Note that version 1.2 of that API has a
- * <a href="https://code.google.com/p/protobuf-java-format/issues/detail?id=47">bug</a>
+ * [bug](https://code.google.com/p/protobuf-java-format/issues/detail?id=47)
  * in the way it serializes unknown fields, so we use our own approach for this case.
  */
-public final class WireTypeAdapterFactory implements TypeAdapterFactory {
-  @SuppressWarnings("unchecked")
-  @Override public @Nullable <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
-    if (type.getRawType().equals(ByteString.class)) {
-      return (TypeAdapter<T>) new ByteStringTypeAdapter();
+class WireTypeAdapterFactory : TypeAdapterFactory {
+  override fun <T> create(gson: Gson, type: TypeToken<T>): TypeAdapter<T>? {
+    if (type.rawType == ByteString::class.java) {
+      return ByteStringTypeAdapter() as TypeAdapter<T>
     }
-    if (Message.class.isAssignableFrom(type.getRawType())) {
-      return (TypeAdapter<T>) new MessageTypeAdapter(gson, type);
-    }
-    return null;
+    return if (Message::class.java.isAssignableFrom(type.rawType)) {
+      MessageTypeAdapter<Nothing, Nothing>(gson, type as TypeToken<Nothing>) as TypeAdapter<T>
+    } else null
```

Comment:
nit: easy to miss this null.  worth keeping it on its own line/return?

Reply:
Wanted to fold those ifs, will do

Reply:
Fixed